nav2_costmap_2d/plugins/costmap_filters/keepout_filter.cpp

```diff
@@ -276,9 +277,9 @@ void KeepoutFilter::process(
         msk_wx = gl_wx;
         msk_wy = gl_wy;
       }
-      // Get mask coordinates corresponding to (i, j) point at mask_costmap_
-      if (mask_costmap_->worldToMap(msk_wx, msk_wy, mx, my)) {
-        data = mask_costmap_->getCost(mx, my);
+      // Get mask coordinates corresponding to (i, j) point at filter_mask_
+      if (worldToMask(filter_mask_, msk_wx, msk_wy, mx, my)) {
+        data = getMaskCost(filter_mask_, mx, my);
```

Comment:
Was the previous object already having its values transformed both in coordinates (worldToMask) and in value (getMaskCost)? Just making sure this is a 1:1 correct reinterpretation.

Reply:
Yes, previous `mask_costmap_` was an `Costmap2D` object made from input `OccupancyGrid` mask message. This constructor [copies](https://github.com/ros-planning/navigation2/blob/63b690a27a4ba1bde24ebd8e6c592c9894eec169/nav2_costmap_2d/src/costmap_2d.cpp#L68-L72) all origin, size, frame, etc... information from occupancy grid, what now is `filter_mask_`, so `worldToMask()` will return the same coordinates for `filter_mask_` as `worldToMap()` for `filter_mask_`. The same story for `getMaskCost()` v.s. `getCost()` return value.

So, this should be 1:1 conversion. This also confirms by keepout test which keeps going after patch, which has a [case](https://github.com/ros-planning/navigation2/blob/63b690a27a4ba1bde24ebd8e6c592c9894eec169/nav2_costmap_2d/test/unit/keepout_filter_test.cpp#L452) for mask and map are in different frames and have different size.